src/Symfony/Component/Notifier/Bridge/Mercure/MercureTransportFactory.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\Mercure;
+
+use Psr\Container\ContainerInterface;
+use Symfony\Component\Notifier\Exception\UnsupportedSchemeException;
+use Symfony\Component\Notifier\Transport\AbstractTransportFactory;
+use Symfony\Component\Notifier\Transport\Dsn;
+use Symfony\Component\Notifier\Transport\TransportInterface;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * @author Mathias Arlaud <mathias.arlaud@gmail.com>
+ *
+ * @experimental in 5.3
+ */
+final class MercureTransportFactory extends AbstractTransportFactory
+{
+    private $publisherLocator;
+
+    public function __construct(EventDispatcherInterface $dispatcher = null, HttpClientInterface $client = null, ContainerInterface $publisherLocator = null)
```

Comment:
If I understand correctly `$publisherLocator` is mandatory:
```suggestion
    public function __construct(ContainerInterface $publisherLocator, EventDispatcherInterface $dispatcher = null, HttpClientInterface $client = null)
```

Reply:
Yes indeed, and actually you pointed out a bigger problem. `MercureTransportFactory` can't be instancied by `Transport` in a static context.

If you have a look at `Notifier/Transport::getDefaultFactories`, you'll see that this doesn't work with Mercure because we don't give the locator to the factory. 

One approach could be to pass the locator in `fromDsn` and `fromDsns` arguments but I'm not a big fan of it... Another idea could be to simply ignore the Mercure factory in a static context but I'm not liking it either. 

To be honest I have no idea of what to do.
Do you have any idea to solve this?

Reply:
What would be the impact if we don't add `MercureTransportFactory` to the default factories (i.e. `Transport::FACTORY_CLASSES`)? 
It seems some factories are already missing from that list (Linkedin and GoogleChat). 
Does it make the transport unusable in some case?  /cc @fabpot

Reply:
As there are precedents, this should be discussed elsewhere. 
Let's remove the transport from `Transport::FACTORY_CLASSES` and move forward here.

Reply:
Agreed!